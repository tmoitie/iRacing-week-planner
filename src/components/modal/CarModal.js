// @flow

import * as React from 'react';
import { useTranslation } from 'react-i18next';
import classnames from 'classnames';
import StarIcon from '../icon/StarIcon';
import Modal from './Modal';

import raceListingStyles from '../styles/raceListing.module.scss';

import allCars from '../../data/cars.json';
import styles from '../../styles/main.module.scss';

type Props = {
  onClose: () => void,
  isOpen: boolean,
  ownedCars: Array<number>,
  favouriteCars: Array<number>,
  carIds: Array<number>,
  seriesName: string,
  seriesId: number,
};

export default function CarModal({ onClose, ownedCars, favouriteCars, isOpen, carIds, seriesName, seriesId }: Props) {
  const cars = [...allCars].sort((a, b) => a.name.localeCompare(b.name)).filter((car) => carIds.includes(car.sku));
  const { t } = useTranslation();

  return (
    <Modal
      id={`modal-cars-${seriesId}`}
      isOpen={isOpen}
      onClose={onClose}
      title={t('Cars for {{series}}', {
        series: t(seriesName),
      })}
      doneAction={onClose}
    >
      <div className={styles['container-fluid']}>
        <div className={styles['table-responsive']}>
          <table className={styles.table} style={{ fontSize: '0.8em' }}>
            <thead>
              <tr>
                <th>{t('Car')}</th>
              </tr>
            </thead>
            <tbody>
              {cars.map((car) => (
                <tr
                  key={car.name}
                  className={classnames({
                    [styles.success]: ownedCars.includes(car.sku),
                    [raceListingStyles.clickableCell]: true,
                  })}
                >
                  <td>
                    {favouriteCars.includes(car.sku) ? (
                      <StarIcon />
                    ) : null}
                    <span> </span>
                    {t(car.name)}
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    </Modal>
  );
}
