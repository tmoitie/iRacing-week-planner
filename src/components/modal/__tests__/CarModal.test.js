import { describe, test } from '@jest/globals';
import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import CarModal from '../CarModal';

import '../../../data/cars.json';

jest.mock('../../../data/cars.json');

describe('components/modal/CarModal', () => {
  test('renders correctly', async () => {
    const user = userEvent.setup();
    const onClose = jest.fn();
    render(
      <CarModal
        onClose={onClose}
        ownedCars={[10494, 10410]}
        favouriteCars={[10461, 10410]}
        isOpen
        carIds={[10410, 10494, 10461, 10405]}
        seriesName="Audi Spectacular"
        seriesId={123}
      />,
    );

    expect(await screen.findByTestId('modal-cars-123')).toMatchSnapshot();

    await user.click(screen.getByText('Close'));
    expect(onClose).toHaveBeenCalled();
  });
});
