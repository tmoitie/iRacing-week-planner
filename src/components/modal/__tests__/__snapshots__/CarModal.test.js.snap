// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/modal/CarModal renders correctly 1`] = `
<div
  class="Modal__Bootstrap modal-dialog modal-lg"
  data-testid="modal-cars-123"
>
  <div
    class="modal-content"
  >
    <div
      class="modal-header"
    >
      <button
        aria-label="Close"
        class="close"
        type="button"
      >
        <span
          class="glyphicon glyphicon-remove"
        />
      </button>
      <h4
        class="modal-title"
      >
        Cars for Audi Spectacular
      </h4>
    </div>
    <div
      class="modal-body"
      style="max-height: 55vh; overflow-y: auto;"
    >
      <div
        class="container-fluid"
      >
        <div
          class="table-responsive"
        >
          <table
            class="table"
            style="font-size: 0.8em;"
          >
            <thead>
              <tr>
                <th>
                  Car
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="success clickableCell"
              >
                <td>
                  <span
                    class="glyphicon glyphicon-star"
                    data-testid="StarIcon"
                  />
                  <span>
                     
                  </span>
                  Audi 90 GTO
                </td>
              </tr>
              <tr
                class="clickableCell"
              >
                <td>
                  <span>
                     
                  </span>
                  Audi R8 LMS GT3
                </td>
              </tr>
              <tr
                class="clickableCell"
              >
                <td>
                  <span
                    class="glyphicon glyphicon-star"
                    data-testid="StarIcon"
                  />
                  <span>
                     
                  </span>
                  Audi RS 3 LMS
                </td>
              </tr>
              <tr
                class="success clickableCell"
              >
                <td>
                  <span>
                     
                  </span>
                  BMW M4 GT3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      class="modal-footer"
    >
      <button
        class="btn btn-primary"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
